mathesar/utils/models.py

```diff
@@ -43,3 +44,15 @@ def update_sa_schema(schema, validated_data):
     if errors:
         raise ValidationError(errors)
     alter_schema(schema.name, schema._sa_engine, validated_data)
+
+
+def update_id_to_attnum(table, validated_data):
+    if 'columns' in validated_data:
+        data = validated_data.get('columns')
+        for column_data in data:
+            col_id = column_data.get('id', None)
+            if col_id is not None:
+                attnum = table.columns.get(id=col_id).attnum
```

Comment:
It would be nice if column id's can be fetched from the database in a single query instead of having to fetch from the db for each column.